data/migrations/20190719084726_projects.js

```diff
@@ -0,0 +1,43 @@
+
+    exports.up = function(knex) {
+        return knex.schema
+            .createTable("projects", tbl => {
+                tbl.increments()
+                tbl
+                .string("name")
+                .unique()
+                .notNullable()
+                tbl
+                .string("description")
+                .notNullable()
+                tbl
+                .boolean("completed")
+                .notNullable()
+            })
+            .createTable("actions", tbl => {
+                tbl.increments()
+                tbl
+                .integer("project_id")
+                .unsigned()
+                .notNullable()
+                .references("id")
+                .inTable("projects")
+                .onDelete('RESTRICT')
+                .onUpdate('CASCADE')
+                tbl
+                .string("description")
+                .notNullable()
+                tbl
+                .string("notes")
+                tbl
+                .boolean("completed")
+                .notNullable()
+            })
+    };
+    
+    exports.down = function(knex) {
+        return knex.schema
+        .dropTableIfExists('projects')
+        .dropTableIfExists('projects_actions')
```

Comment:
In the case of a one to many relationship, where an action is unique to a project, you would not required a third table `project_actions` like you were thinking. In the case of the many to many relationship, where an action can be used with multiple projects, you would be creating the third table and would also remove the foreign key relationship from the `actions` table. Nice work with the first case though, this is exactly what we wanted you to have for this project.